Signal/src/UIView+OWS.m

```diff
@@ -54,22 +60,33 @@ - (void)autoPinWidthToSuperview
     return result;
 }
 
-- (void)autoPinHeightToSuperviewWithMargin:(CGFloat)margin
+- (NSArray<NSLayoutConstraint *> *)autoPinHeightToSuperviewWithMargin:(CGFloat)margin
 {
-    [self autoPinEdge:ALEdgeTop toEdge:ALEdgeTop ofView:self.superview withOffset:+margin];
-    [self autoPinEdge:ALEdgeBottom toEdge:ALEdgeBottom ofView:self.superview withOffset:-margin];
+    NSArray<NSLayoutConstraint *> *result = @[
+        [self autoPinEdge:ALEdgeTop toEdge:ALEdgeTop ofView:self.superview withOffset:+margin],
+        [self autoPinEdge:ALEdgeBottom toEdge:ALEdgeBottom ofView:self.superview withOffset:-margin],
+    ];
+    return result;
 }
 
-- (void)autoPinHeightToSuperview
+- (NSArray<NSLayoutConstraint *> *)autoPinHeightToSuperview
 {
-    [self autoPinEdge:ALEdgeTop toEdge:ALEdgeTop ofView:self.superview];
-    [self autoPinEdge:ALEdgeBottom toEdge:ALEdgeBottom ofView:self.superview];
+    NSArray<NSLayoutConstraint *> *result = @[
+        [self autoPinEdge:ALEdgeTop toEdge:ALEdgeTop ofView:self.superview],
+        [self autoPinEdge:ALEdgeBottom toEdge:ALEdgeBottom ofView:self.superview],
+    ];
+    return result;
 }
 
-- (void)autoPinToSuperviewEdges
+- (NSArray<NSLayoutConstraint *> *)autoPinToSuperviewEdges
```

Comment:
`autoPinEdgesToSuperviewEdges` uses leading/trailing, not left right so it will respect margin insets.  I agree that this is confusing but I don't want to address it in this PR.

Reply:
https://trello.com/c/wlQCiMlT/1720-reconcile-autopinedgestosuperviewedges-and-autopintosuperviewedges

Reply:
I made a card in not yet prioritized.